### PR TITLE
dda-33: Change temporary password form and functionality

### PR DESCRIPTION
**Pull request by shanben7**

### CODE REVIEW

**Comment by shanben7** [comment]
@pritpalc screenshots are here, sorry pasted them in issue

https://github.com/CodetheChangeFoundation/dda/issues/33

File: navigation/AppNavigator.js [hunk 1]
```diff
@@ -3,15 +3,26 @@ import { createAppContainer, createSwitchNavigator } from 'react-navigation';
...
+
+
+
+
+
+
+
     EmployerSignUp: EmployerSignUpScreen,
+
+
```
Comment (pritpalc) [inline navigation/AppNavigator.js]: Could you remove all of this whitespace

File: screens/auth/AuthScreen.js [hunk 2]
```diff
@@ -24,14 +23,18 @@ export default class AuthScreen extends React.Component {
         this.setState({[field]: value});
     }
 
-    render() {
+    render(url) {
```
Comment (pritpalc) [inline screens/auth/AuthScreen.js]: It's better practice not having the `render` function take in an argument, you can instead use a state variable for this.

File: screens/auth/ChangePasswordScreen.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        this.handleFormChange = this.handleFormChange.bind(this);
+    }
+
+    handleFormChange(field, value) {
+        this.setState({[field]: value});
+    }
+
+    handlePasswordChange = () => {
+        const {oldPassword, newPassword, confirmNewPassword} = this.state;
+        const email = this.state.email;
```
Comment (pritpalc) [inline screens/auth/ChangePasswordScreen.js]: Is there a reason why you're not combining this line with the line above it?

Comment (shanben7) [reply screens/auth/ChangePasswordScreen.js]: woops, nope i'll merge them

**Comment by shanben7** [comment]
Screen Shot 2019-06-18 at 1 11 34 PM
as per your comment on #33 i added some error handling, it opens an Alert window and displays an error message. When the user clicks OK on the alert, they can re-fill out the form.

This covers error cases:
- password does not conform to requirements
- they enter an email that doesn't exist in our db
- they enter the wrong password
- the 2 fields for their new password don't match
- they've already successfully changed their temporary pw

**Pull request merged**